Yank/utils.py

```diff
@@ -235,6 +235,9 @@ class CombinatorialLeaf(list):
     """List type that can be expanded combinatorially in CombinatorialTree."""
     def __repr__(self):
         return "Combinatorial({})".format(super(CombinatorialLeaf, self).__repr__())
+    def __eq__(self, other):
+        # Order does not mater, so comparisons should not rely on them
+        return collections.Counter(self) == collections.Counter(other)
```

Comment:
Does the order of the `CombinatorialLeaf` matter when making equality checks?

`CombinatorialLeaf`s generated from `CombinatorialTree`s can have lists pop out in any order, causing equality tests to sometimes fail. I overloaded the equality operator for the `CombinatorialLeaf` instead of making it a list, it makes a type of `set`. (`set` might be a simpler cast here over `collections.Counter`, but it works for now).